src/sdk/lib/StorageManager.js

```diff
@@ -108,3 +108,16 @@ export class LocalStorageManager extends BaseStorageManager
         super( storageName, window.localStorage );
     }
 }
+
+export const getStorageClass = (opts) => {
+    // Only cookie support domain based storage
+    if (opts.storageType === 'cookie' || opts.cookieDomain) {
+        return class CookieStorageManager extends BaseStorageManager {
```

Comment:
can you formally draw this out as a separate CookieStorageManager not as an in-line class?
you can have CookieStorageManager having a different constructor and still pass the domain in